contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,367 @@
-      - [Best Effort](#best-effort)
-    - [With Systemd](#with-systemd)
-    - [Hierarchy Outline](#hierarchy-outline)
-      - [QoS Policy Design Decisions](#qos-policy-design-decisions)
-  - [Implementation Plan](#implementation-plan)
-      - [Top level Cgroups for QoS tiers](#top-level-cgroups-for-qos-tiers)
-      - [Pod level Cgroup creation and deletion (Docker runtime)](#pod-level-cgroup-creation-and-deletion-docker-runtime)
-      - [Container level cgroups](#container-level-cgroups)
-      - [Rkt runtime](#rkt-runtime)
-      - [Add Pod level metrics to Kubelet's metrics provider](#add-pod-level-metrics-to-kubelets-metrics-provider)
-  - [Rollout Plan](#rollout-plan)
-  - [Implementation Status](#implementation-status)
-
-<!-- END MUNGE: GENERATED_TOC -->
+# Kubelet pod level resource management
+
+**Authors**:
+
+1. Buddha Prakash (@dubstack)
+1. Vishnu Kannan (@vishh)
+1. Derek Carr (@derekwaynecarr)
+
+**Last Updated**: 01/30/2017
+
+**Status**: Implementation planned for Kubernetes 1.6
+
+This document proposes a design for introducing pod level resource accounting
+to Kubernetes. It outlines the implementation and associated rollout plan.
 
 ## Introduction
 
-As of now [Quality of Service(QoS)](../../docs/design/resource-qos.md) is not enforced at a pod level. Excepting pod evictions, all the other QoS features are not applicable at the pod level.
-To better support QoS, there is a need to add support for pod level resource accounting in Kubernetes.
+Kubernetes supports container level isolation by allowing users
+to specify [compute resource requirements](resources.md) via requests and
+limits on individual containers.  The `kubelet` delegates creation of a
+cgroup sandbox for each container to its associated container runtime.
+
+Each pod has an associated [Quality of Service (QoS)](resource-qos.md)
+class based on the aggregate resource requirements made by individual
+containers in the pod.  The `kubelet` has the ability to
+[evict pods](kubelet-eviction.md) when compute resources are scarce. It evicts
+pods with the lowest QoS class in order to attempt to maintain stability of the
+node.
+
+The `kubelet` has no associated cgroup sandbox for individual QoS classes or
+individual pods.  This inhibits the ability to perform proper resource
+accounting on the node, and introduces a number of code complexities when
+trying to build features around QoS.
+
+This design introduces a unified cgroup hierarchy to enable the following:
+
+1. Improve enforcement of QoS class on the node.
+1. Simplify resource accounting at the pod level.
+1. Allow containers in a pod to share slack resources.  A pod with a Burstable
+container and a BestEffort container is classified as a Burstable pod.  The
+BestEffort container is not able to consume slack resources from the sibling
+Burstable container.  It must instead compete for scarce resources at the node
+level across all containers in all QoS classes which violates the design.
```

Comment:
that is correct.  i had taken this verbiage from the original document.  will update that it will allow containers in a pod to share slack resources within its qos tier.